### DISCUSSION TITLE
How do I set the functions to use GPU?

### DISCUSSION THREAD

**Discussion by KuriaMaingi** [Filimoa/open-parse#65] [General]
I installed openparse using pip install openparse[ml]

I have a table extraction (using unitable) that is very long running but exclusively using the CPU. I noticed two things:
- Your note to run it [using GPU ](https://github.com/Filimoa/open-parse/blob/984ed684ecbace04bba60b85d4349ffe54ea3504/docs/processing/parsing-tables/unitable.md?plain=1#L44)
- This note about setting [CPU in the config ](https://github.com/Filimoa/open-parse/blob/984ed684ecbace04bba60b85d4349ffe54ea3504/docs/config.md)

My question is this:
- If I want to use the GPU, is it as simple as openparse.config.set_device("cuda")?
- Do I have to separately install pytorch in my venv as it didn't appear to be a linked dependency? Any guidance on the min/max version especially of CUDA?

**Comment by Filimoa** [comment]
Correct on the just updating the config.

I haven't pinned a minimum version of pytorch yet - should work on 1X and 2X. Honestly not sure which version it'd break on
